### PR TITLE
[CP-1282] User data migration

### PR DESCRIPTION
**Pull request by SP2FET**
**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible
- [x] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

**Review by swichu91 — APPROVED** [review]

File: cmake/modules/AddDirectories.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    endforeach ()
+
+    add_custom_target(
+            ${_ASSETS_TARGET}
+            DEPENDS ${_ASSETS_DEPENDS}
+            ${command}
+            COMMENT
+            "Adding directories: ${_ASSETS_DIRECTORIES}"
+    )
+endfunction()
```
Comment (kkleczkowski) [inline cmake/modules/AddDirectories.cmake:31]: Missing new line

File: products/BellHybrid/paths/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+add_library(paths Paths.cpp)
+add_library(bell::paths ALIAS paths)
+
+target_include_directories(paths PUBLIC .)
+
+target_link_libraries(paths PRIVATE module-vfs)
```
Comment (kkleczkowski) [inline products/BellHybrid/paths/CMakeLists.txt:6]: Missing new line.

File: products/BellHybrid/services/db/ServiceDB.cpp [hunk 3]
```diff
@@ -99,20 +71,4 @@ sys::ReturnCodes ServiceDB::InitHandler()
...
-
-    return std::all_of(databaseAgents.cbegin(), databaseAgents.cend(), [&backupPath](const auto &agent) {
-        if (not agent->storeIntoFile(backupPath / std::filesystem::path(agent->getDbFilePath()).filename())) {
-            LOG_ERROR("%s backup failed", agent->getAgentName().c_str());
-            return false;
-        }
-        return true;
-    });
-}
+}
```
Comment (kkleczkowski) [inline products/BellHybrid/services/db/ServiceDB.cpp:74]: Missing new line.

File: products/BellHybrid/services/db/databases/databases.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "BellHybrid": {
+    "databases":[
+      {"name": "events", "version": "0"},
+      {"name": "multimedia", "version": "0"},
+      {"name": "meditation_stats", "version": "0"},
+      {"name": "settings_bell", "version": "0"}
+    ]
+  }
+}
```
Comment (kkleczkowski) [inline products/BellHybrid/services/db/databases/databases.json:10]: Missing new line.

File: products/PurePhone/services/db/databases/databases.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+      {"name": "contacts", "version": "0"},
+      {"name": "custom_quotes", "version": "0"},
+      {"name": "notes", "version": "0"},
+      {"name": "notifications", "version": "0"},
+      {"name": "predefined_quotes", "version": "0"},
+      {"name": "settings_v2", "version": "0"},
+      {"name": "sms", "version": "0"}
+    ]
+  }
+}
```
Comment (kkleczkowski) [inline products/PurePhone/services/db/databases/databases.json:17]: Missing new line.

File: scripts/lua/install.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    cp *.lua ${DESTINATION}
+
+    if [ -n "$3" ]; then
+        echo "Generating UDM update package"
+        cp update_udm.lua ${DESTINATION}/update.lua
+        cp products/${PRODUCT}/update_product.lua ${DESTINATION}
+    fi
+    rm ${DESTINATION}/update_udm.lua
+    cd -
+fi
```
Comment (kkleczkowski) [inline scripts/lua/install.sh:43]: Missing new line.

File: scripts/lua/migration/test/install_dependencies.sh [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+        echo "$1 command could not be found. Please install it using system package manager"
+        exit 1
+    fi
+}
+
+if_cmd_installed "luarocks"
+
+luarocks install lsqlite3complete 
+luarocks install lfs
+luarocks install luaunit
```
Comment (kkleczkowski) [inline scripts/lua/migration/test/install_dependencies.sh:16]: Missing new line.

File: cmake/modules/AddDirectories.cmake (hunk 1 above)
Comment (SP2FET) [reply cmake/modules/AddDirectories.cmake]: fixed

**Review by kkleczkowski — APPROVED** [review]
GH marked some files as empty. Please double check if the empty files are needed.

**Review by Lefucjusz — APPROVED** [review]

File: cmake/modules/AddScripts.cmake [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+    add_custom_target(
+            ${_ARG_TARGET}
+            DEPENDS ${_ARG_DEPENDS}
+
+            COMMAND  ${PROJECT_SOURCE_DIR}/scripts/lua/install.sh ${_ARG_PRODUCT} ${_ARG_DEST_DIR} udm
+            COMMENT
+            "Installing scripts for ${_ARG_PRODUCT} to ${_ARG_DEST_DIR} directory"
+    )
+endfunction()
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply cmake/modules/AddScripts.cmake:18]: fixed

File: cmake/modules/DownloadAsset.cmake [hunk 9]
```diff
@@ -18,8 +18,7 @@ function(download_asset_release asset_name_in asset_name_out asset_repo asset_ve
 
     add_custom_target(${asset_name_out}-target DEPENDS ${asset_repo})
 
-    multicomp_install(PROGRAMS ${CMAKE_BINARY_DIR}/${asset_repo} DESTINATION "./"
-        COMPONENTS Standalone Update)
+#    multicomp_install(PROGRAMS ${CMAKE_BINARY_DIR}/${asset_repo} DESTINATION "./" COMPONENTS Standalone Update)
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply cmake/modules/DownloadAsset.cmake:21]: fixed

File: module-apps/application-desktop/windows/ClosingWindow.hpp [hunk 10]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply module-apps/application-desktop/windows/ClosingWindow.hpp]: fixed

File: image/CMakeLists.txt [hunk 11]
```diff
@@ -21,16 +21,10 @@ endif()
...
 )
 
-multicomp_install(
-    FILES
-        ${ASSETS_DEST_DIR}/.boot.json
-        ${ASSETS_DEST_DIR}/.boot.json.crc32
-    DESTINATION ./
-    COMPONENTS Standalone Update
-)
+#multicomp_install(FILES  ${USER_DEST_DIR}/boot.json DESTINATION ./ COMPONENTS Standalone Update)
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply image/CMakeLists.txt]: fixed

File: module-utils/tar/tar.cpp [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+            const auto full_path = where / entry.name();
+            if (entry.is_directory() and isValidDirectoryEntry(entry)) {
+                std::filesystem::create_directories(full_path);
+            }
+            else if (entry.is_file()) {
+                write_to_file(entry, full_path);
+            }
+        }
+    }
+} // namespace tar
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply module-utils/tar/tar.cpp]: fixed

File: products/BellHybrid/alarms/include/AlarmSoundPaths.hpp [hunk 13]
```diff
@@ -18,4 +19,4 @@ namespace alarms::paths
 
     /// Check if system paths exist. In case of error, returns vector of missing entries.
     std::vector<std::filesystem::path> validate() noexcept;
-} // namespace alarms::paths
+} // namespace alarms::paths
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply products/BellHybrid/alarms/include/AlarmSoundPaths.hpp]: fixed

File: products/BellHybrid/alarms/src/AlarmSoundPaths.cpp [hunk 14]
```diff
@@ -68,4 +68,4 @@ namespace alarms::paths
         }
         return ret;
     }
-} // namespace alarms::paths
+} // namespace alarms::paths
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply products/BellHybrid/alarms/src/AlarmSoundPaths.cpp]: fixed

File: products/BellHybrid/paths/CMakeLists.txt (hunk 2 above)
Comment (SP2FET) [reply products/BellHybrid/paths/CMakeLists.txt]: fixed

File: products/BellHybrid/services/db/ServiceDB.cpp (hunk 3 above)
Comment (SP2FET) [reply products/BellHybrid/services/db/ServiceDB.cpp]: fixed

File: products/BellHybrid/services/db/databases/databases.json (hunk 4 above)
Comment (SP2FET) [reply products/BellHybrid/services/db/databases/databases.json]: fixed

File: products/BellHybrid/services/db/tests/CMakeLists.txt [hunk 15]
```diff
@@ -6,5 +6,5 @@ add_catch2_executable(
 
         LIBS
         bell::db::meditation_stats
-        USE_FS
+        module-db::test::helpers
 )
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply products/BellHybrid/services/db/tests/CMakeLists.txt]: fixed

File: products/PurePhone/services/db/databases/databases.json (hunk 5 above)
Comment (SP2FET) [reply products/PurePhone/services/db/databases/databases.json]: fixed

File: scripts/lua/install.sh (hunk 6 above)
Comment (SP2FET) [reply scripts/lua/install.sh]: fixed

File: scripts/lua/migration/test/install_dependencies.sh (hunk 7 above)
Comment (SP2FET) [reply scripts/lua/migration/test/install_dependencies.sh:16]: fixed

**Pull request merged**